packages/vite/src/node/server/moduleGraph.ts

```diff
@@ -150,15 +150,26 @@ export class ModuleGraph {
     const prevImports = mod.importedModules
     const nextImports = (mod.importedModules = new Set())
     let noLongerImported: Set<ModuleNode> | undefined
+    let resolvePromises = []
     // update import graph
     for (const imported of importedModules) {
-      const dep =
-        typeof imported === 'string'
-          ? await this.ensureEntryFromUrl(imported, ssr)
-          : imported
-      dep.importers.add(mod)
-      nextImports.add(dep)
+      if (typeof imported === 'string') {
+        resolvePromises.push(
+          this.ensureEntryFromUrl(imported, ssr).then((dep) => {
+            dep.importers.add(mod)
+            nextImports.add(dep)
+          }),
+        )
+      } else {
+        imported.importers.add(mod)
+        nextImports.add(imported)
```

Comment:
I think it would be a good idea to keep the `nextImports` in the same order as before, at least to help with debugging and to make the runs more stable.

We could await for the result of `Promise.all` and then add that to nextImports in order.

Reply:
done